internal/store/store.go

```diff
@@ -42,27 +43,27 @@ func NewBlockStore(db dbm.DB) *BlockStore {
 }
 
 // Base returns the first known contiguous block height, or 0 for empty block stores.
-func (bs *BlockStore) Base() int64 {
+func (bs *BlockStore) Base() (int64, error) {
 	iter, err := bs.db.Iterator(
 		blockMetaKey(1),
 		blockMetaKey(1<<63-1),
 	)
 	if err != nil {
-		panic(err)
+		return 0, pkg_errors.Wrap(err, "Error while iterating blockMetaKeys")
 	}
 	defer iter.Close()
 
 	if iter.Valid() {
 		height, err := decodeBlockMetaKey(iter.Key())
 		if err == nil {
-			return height
+			return height, nil
 		}
 	}
 	if err := iter.Error(); err != nil {
 		panic(err)
 	}
 
-	return 0
+	return 0, nil
```

Comment:
I wonder if it's probably safer to return an error here as it means the block store is empty, although 0 is an invalid height, we might not be checking for that in every case that this function is called. Same can be applied to `Height`. @tychoish what do you think?

Reply:
Definitely, return an error here.